llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -3496,21 +3496,26 @@ static SDValue matchSplatAsGather(SDValue SplatVal, MVT VT, const SDLoc &DL,
   if (SplatVal.getOpcode() != ISD::EXTRACT_VECTOR_ELT)
     return SDValue();
   SDValue Vec = SplatVal.getOperand(0);
-  // Only perform this optimization on vectors of the same size for simplicity.
   // Don't perform this optimization for i1 vectors.
   // FIXME: Support i1 vectors, maybe by promoting to i8?
-  if (Vec.getValueType() != VT || VT.getVectorElementType() == MVT::i1)
+  if (VT.getVectorElementType() == MVT::i1)
     return SDValue();
   SDValue Idx = SplatVal.getOperand(1);
   // The index must be a legal type.
   if (Idx.getValueType() != Subtarget.getXLenVT())
     return SDValue();
 
+  if (Vec.getValueSizeInBits().getKnownMinValue() >
```

Comment:
Is it possible to have a mix of fixed and scalable vector type here such that comparing the known minimum value is meaningless?

Reply:
I changed it to use `TypeSize::isKnownLE` negated, so we also bail if we are comparing scalable and fixed vectors

Reply:
The check in isKnownLE allows LHS to be fixed and RHS to be scalable. I haven't yet checked if that's ok here.

Reply:
I think that should be fine because vscale has to be at least one. Or are you talking about if checking Vec <= VT is enough?